src/com/yetanalytics/lrs_admin_ui/input.cljs

```diff
@@ -0,0 +1,29 @@
+(ns com.yetanalytics.lrs-admin-ui.input
+  (:require [cljs.spec.alpha :as s :include-macros true]
+            [com.yetanalytics.lrs-admin-ui.functions.password :as pass]
+            [clojure.set :refer [intersection]]))
+
+;; Minimum lengths
+(def p-min-len 10)
+(def u-min-len 7)
+
+(s/def :valid-account/password
+  (let [p-contains? (fn [pass chars]
+                      (not-empty (intersection (set pass)
+                                               (set chars))))]
+    (s/and string?
+           #(>= (count %) p-min-len)
+           #(p-contains? % pass/digit-chars)
+           #(p-contains? % pass/upper-chars)
+           #(p-contains? % pass/lower-chars)
+           #(p-contains? % pass/special-chars))))
```

Comment:
1. Is there a reason why `p-contains?` is not a separate private function?
2. Definitely make separate `(def char-set (set chars))` defs since `set` is not the most cheap operation.
3. After some playing around, the following code is faster than calling `intersection`:
```clojure
(every? char-set pass)
```
I get about a 15-fold speedup with this.

Reply:
not particularly i can pull it out, and sure that makes sense abt sets, better to do once.

Reply:
Also after some playing out, I found that this:
```clojure
(some char-set pass)
```
is around 5-6 times faster than the original:
```clojure
(not-empty (intersection char-set (set pass))
```
Sure it's not really the most important place to optimize but it's a pretty easy one to implement so why not.